### ISSUE TITLE
"NuxtPage / NuxtLayout component has not been used" warn when NuxtLayout async show up

### ISSUE THREAD

**Issue by serkodev** [nuxt/nuxt#25912]
### Environment

- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.10.2
- CLI Version:      3.10.1
- Nitro Version:    2.8.1
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-2vasos?file=app.vue

### Describe the bug

I used `<NuxtPage />` and `<NuxtLayout />` components in the below way:
```vue
    <div v-if="loading">loading...</div>
    <NuxtLayout v-else>
      <NuxtPage />
    </NuxtLayout>
```

But it shows up the below warning in console log.
```
[nuxt] Your project has pages but the `<NuxtPage />` component has not been used. You might be using the `<RouterView />` component instead, which will not work correctly in Nuxt. You can set `pages: false` in `nuxt.config` if you do not wish to use the Nuxt `vue-router` integration.
[nuxt] Your project has layouts but the `<NuxtLayout />` component has not been used.
```

### Additional context

My purpose is fetching some environment data from server before NuxtLayout and NuxtPage rendered and shows a loading layout.

ps. I disabled the `ssr` options.

**Comment by manniL** [comment]
Yup, that's not detected by the current warning process. You can ignore that one for now 😋

**Comment by abarke** [comment]
I'm also getting this error in a Nuxt Ionic App. In this case I am not using `NuxtPage`, because I'm using `ion-router` which I assume uses `NuxtPage`. Any way to disable this false positive warning?

![image](https://github.com/nuxt/nuxt/assets/34715541/b2936fa3-80b8-404b-a362-a2af8a96a572)

**Comment by rodrigogs** [comment]
I have this same warning when using Amplify Authenticator:

```vue
    <NuxtLayout>
      <Authenticator>
        <NuxtPage />
      </Authenticator>
    </NuxtLayout>
```

**Comment by serkodev** [comment]
@manniL i was wondering if it is possible to add a option to ignore the warning message?

**Comment by amE80** [comment]
![nuxt](https://github.com/nuxt/nuxt/assets/101411848/93cfe13b-0efe-46f1-bc71-f514815fce67)

I had the same warning in my console, I separated the files (in the picture) and the warning was gone and I could use my pages.

**Comment by amE80** [comment]
![nuxt](https://github.com/nuxt/nuxt/assets/101411848/1cf4db47-f434-4962-a78c-2a4459a8b3cc)

@serkodev  By separating the files, I could remove the console warning and everything worked great (I had the same issue)

**Comment by Plinpod** [comment]
Same issue. Showing a loader than switching to `<NuxtPage />` anyway to ignore this warning? Or is this considered bad practice?

**Comment by admg** [comment]
Same problem. I'm hiding NuxtPage component and showing login dialog, when user in not authenticated. Is there any other way to conditionally not render any nuxt page, without this warning?

**Comment by steveulfelder** [comment]
The above didn't work for me. This does:

```
<template>
  <NuxtLayout v-if="loaded" :name="layout">
    <NuxtPage />
  </NuxtLayout>

  <!-- Dummy layout to prevent Nuxt warnings while loading -->
  <NuxtLayout v-else :name="false">
    <NuxtPage :name="''" />
  </NuxtLayout>
</template>
```

Do be aware that by waiting to render until your data fetching/client width calculation is complete, you lose one of the main benefits of SSR (fast first paints). May be worth refactoring to a SPA instead of contorting Nuxt like this.

**Comment by serkodev** [comment]
@steveulfelder Thanks for sharing the solution and I totally agree your notice.

However, by using Nuxt's `ssr: false` option to develop an SPA, you can also enjoy Nuxt's excellent DX. The warning still appear when `ssr: false` is set.

**Comment by WojciechxFalkowski** [comment]
When it will be fixed? This issue is opened since February

**Comment by ra-dave** [comment]
Having this issue as well and all recommended workarounds so far have not worked.  Nuxt 3.13.2

**Comment by kpsh** [comment]
I got the same warning and found a possible fix. In my case I was loading `<slot />` only when a `ref()` was true. I think `<slot />` is always required by `<NuxtLayout />` in order to render the content of pages.

The easier solution would be to move your page rendering logic in a middleware or do it manually in each page (not recommended). I don't think conditionally loading the slot is a Nuxt problem, or how its intended to work.

In my project i am using `<USlideover />` with a slot inside from nuxt/ui to display my pages content. My index page needs to be empty so i just load it when im not using the slideover with `<slot v-if="route.path === '/'" />`

**Comment by VNWA** [comment]
<div class="bg-white" v-if="isLoading">
      <Loading />

    </div>
<div class="bg-white" v-if="!isLoading">
      header

    </div>

<slot/>
<div class="bg-white" v-if="!isLoading">
      footer

    </div>

**Comment by Pwntus** [comment]
You must always render `<NuxtPage />`to avoid the warning.

For a loading screen, I found this to work:
```Vue
<template>
  <NuxtLayout>
    <NuxtPage :key="inited ? route.path : 'uninitialized'" />
  </NuxtLayout>
</template>
```
This solution uses a `:key` prop to force a re-render when `inited` changes. When not initialized, uses a static key 'uninitialized' which should prevent any route matching. When initialized, uses the actual route path as the key to ensure proper routing.

**Comment by TheAlexLichter** [comment]
This warning should be an ESLint rule eventually and not a runtime warning as there are edge cases like using v-if / loading only on mounted.

If you have the warning and use one of these cases, you can surely ignore it!

**Comment by sukumaar** [comment]
This is really confusing and false positive warning message. I hope the fix would come as early as possible.

**Comment by anton-gor-dev** [comment]
https://github.com/nuxt/nuxt/pull/35184

I opened a small PR to clarify the warning message for this case.

The current warning says that `<NuxtPage />` / `<NuxtLayout />` have not been used, which can be misleading when they are rendered conditionally and the runtime check runs before the component is mounted.

This PR does not change the detection logic, but updates the wording to better reflect the current behavior.
